### PR TITLE
Convert eager-eval utils to actions

### PR DESCRIPTION
**Pull request by jasonLaster**
wip to be able to fetch the selected frame from the store

### CODE REVIEW

File: src/devtools/client/webconsole/components/Input/EagerEvalFooter.tsx [hunk 1]
```diff
@@ -1,18 +1,21 @@
...
-import { eagerEvaluateExpression } from "../../utils/autocomplete-eager";
+import { eagerEvaluateExpression } from "../../actions/autocomplete-eagerer";
 
 function useEagerEvalPreview(expression: string) {
   const [value, setValue] = useState<ValueFront | null>(null);
+  const dispatch = useDispatch();
+
   const expressionRef = useRef(expression);
 
   useEffect(() => {
```
Comment (jcmorrow) [inline src/devtools/client/webconsole/components/Input/EagerEvalFooter.tsx:13]: The linter wants `dispatch` to be in the deps array here, which I think should be harmless and technically correct (though probably not functionally important).

File: src/devtools/client/webconsole/components/Input/useAutocomplete.tsx [hunk 2]
```diff
@@ -38,7 +41,7 @@ function useGetEvalMatches(expression: string) {
         return;
       }
 
-      const evaluatedProperties = await getEvaluatedProperties(propertyExpression.left);
+      const evaluatedProperties = await dispatch(getEvaluatedProperties(propertyExpression.left));
```
Comment (jcmorrow) [inline src/devtools/client/webconsole/components/Input/useAutocomplete.tsx:44]: Likewise in this one.

File: src/devtools/client/webconsole/actions/autocomplete-eager.ts [hunk 3]
```diff
@@ -0,0 +1,57 @@
+import { getSelectedFrame } from "devtools/client/debugger/src/selectors";
+
+import { ThreadFront, ValueFront } from "protocol/thread";
+import { getPropertiesForObject } from "../utils/autocompleteomplete";
```
Comment (jcmorrow) [inline src/devtools/client/webconsole/actions/autocomplete-eager.ts:4]: 🙃

**Review by jcmorrow — CHANGES_REQUESTED** [review]
The concept looks good to me, but there's a couple of linter things and a typo to fix ⌨️

File: src/devtools/client/webconsole/components/Input/EagerEvalFooter.tsx (hunk 1 above)
Comment (jaril) [reply src/devtools/client/webconsole/components/Input/EagerEvalFooter.tsx:13]: yeah it's [silly](https://react-redux.js.org/api/hooks#usedispatch) 😬

**Pull request closed without merging**